agera/src/main/java/com/google/android/agera/BaseObservable.java

```diff
@@ -71,7 +71,11 @@ public synchronized final void addUpdatable(@NonNull final Updatable updatable)
       if (handler.hasMessages(MSG_LAST_REMOVED, this)) {
         handler.removeMessages(MSG_LAST_REMOVED, this);
       } else {
-        handler.obtainMessage(WorkerHandler.MSG_FIRST_ADDED, this).sendToTarget();
+        if (Looper.myLooper() == handler.getLooper()) {
+          observableActivated();
```

Comment:
I would probably prefer this called outside the `synchronized` block (once you merge the other change you'll have an explicit block to go out of).